go/internal/hashing/hash_string.go

```diff
@@ -0,0 +1,30 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//go:build !go1.20
```

Comment:
Does this work with go1.21 or later?
(I think that the answer is "no".)

Can we use `//go:build <go1.20` or something?

Reply:
the Go build constraints syntax revolves around the existence of "tags" and doesn't support `<` or `>` operators, it only supports parens, `&`, `!`, and `|`. That said, all go version tags are set for all subsequent versions.  So basically `!go1.20` == `<go1.20` and `go1.20` == `>=go1.20`.

So this will do the correct thing and will also exclude go1.21 and later.

Reply:
I see.
Then how about renaming this file to `hash_string_go1.19.go` and `hash_string_go1.20.go` to `hash_string.go`? Because this file is a special case and `hash_string_go1.20.go` is the normal case in future.